Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -326,6 +327,7 @@ async Task DisconnectInternalAsync(Task sender, Exception exception, MqttClientA
             catch (Exception adapterException)
             {
                 _logger.Warning(adapterException, "Error while disconnecting from adapter.");
+                reasonCode = MqttClientDisconnectReason.UnspecifiedError;
```

Comment:
Why do you use "UnspecifiedError" here and below "ImplementationSpecificError"? I am just curious why this is not the same 😄

Reply:
No idea :D Which one should I use? I wasn't really sure about that...

Reply:
I also have no idea 😄 But to me they should be at least equal. My proposal is "UnspecifiedError" for both cases.